perses/samplers/samplers.py

```diff
@@ -1032,138 +371,67 @@ def _geometry_ncmc_geometry(self, topology_proposal, positions, old_log_weight,
         -------
         logP_accept : float
             Log of acceptance probability of entire Expanded Ensemble switch (Eq. 25 or 46)
-        ncmc_new_positions : simtk.unit.Quantity with dimension [natoms, 3] with units of distance.
-            Positions of new atoms at the end of the NCMC switching.
+        ncmc_new_sampler_state : openmmtools.states.SamplerState
+            Configurational properties of new atoms at the end of the NCMC switching.
         """
         if self.verbose: print("Updating chemical state with geometry-ncmc-geometry scheme...")
 
         from perses.tests.utils import compute_potential
 
-        logP_chemical = topology_proposal.logp_proposal
+        logP_chemical_proposal = topology_proposal.logp_proposal
 
-        old_positions = positions
-        initial_reduced_potential = self.sampler.thermodynamic_state.beta * compute_potential(topology_proposal.old_system, old_positions, platform=self.ncmc_engine.platform)
-        logP_initial = -initial_reduced_potential + old_log_weight
-
-        geometry_new_positions, logP_forward = self._geometry_forward(topology_proposal, old_positions)
+        old_thermodynamic_state = self.sampler.thermodynamic_state
+        new_thermodynamic_state = self._system_to_thermodynamic_state(topology_proposal.new_system)
 
-        ncmc_new_positions, ncmc_old_positions, logP_work, logP_energy = self._ncmc_hybrid(topology_proposal, old_positions, geometry_new_positions)
+        initial_reduced_potential = feptasks.compute_reduced_potential(old_thermodynamic_state, sampler_state)
```

Comment:
Why use `feptasks.compute_reduced_potential` and not [`old_thermodynamic_state.reduced_potential(sampler_state)`](https://openmmtools.readthedocs.io/en/latest/api/generated/openmmtools.states.ThermodynamicState.html#openmmtools.states.ThermodynamicState.reduced_potential)?

Reply:
I can't access the docs right now for some reason (timing out), but I think I need to retrieve a context, which is what I do in the `feptasks.compute_reduced_potential` method.